File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int,int> hashMap; 
        for(int i = 0; i < nums.size(); i++){
            if(hashMap.count(nums[i])){
                if(abs(hashMap[nums[i]] - i) <= k) return true; 
            }

            hashMap[nums[i]] = i; 
        }
        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer> hashMap = new HashMap<>(); 
        for(int i = 0; i < nums.length; i++){
            int curr = nums[i]; 
            if(hashMap.containsKey(curr) && Math.abs(hashMap.get(curr) - i) <= k) return true; 
            hashMap.put(curr,i); 
        }
        return false; 
    }
}